WordPress/Classes/ViewRelated/Blog/CreateNewBlog/CreateNewBlogViewController.m

```diff
@@ -0,0 +1,551 @@
+    
+    // In order to fit controls ontol all phones, the textField height is smaller on iPhones
+    // versus iPads.
+    CGFloat textFieldHeight = IS_IPAD ? CreateBlogTextFieldHeight: CreateBlogTextFieldPhoneHeight;
+    
+    // Layout Site Title
+    x = (viewWidth - CreateBlogTextFieldWidth)/2.0;
+    y = CGRectGetMaxY(_titleLabel.frame) + CreateBlogStandardOffset;
+    _siteTitleField.frame = CGRectIntegral(CGRectMake(x, y, CreateBlogTextFieldWidth, textFieldHeight));
+    
+    // Layout Site Address
+    x = (viewWidth - CreateBlogTextFieldWidth)/2.0;
+    y = CGRectGetMaxY(_siteTitleField.frame) - 1;
+    _siteAddressField.frame = CGRectIntegral(CGRectMake(x, y, CreateBlogTextFieldWidth, textFieldHeight));
+    
+    // Layout WordPressCom Label
+    [_siteAddressWPComLabel sizeToFit];
+    CGSize wordPressComLabelSize = _siteAddressWPComLabel.frame.size;
+    wordPressComLabelSize.height = _siteAddressField.frame.size.height - 10;
+    wordPressComLabelSize.width += 10;
+    _siteAddressWPComLabel.frame = CGRectMake(_siteAddressField.frame.size.width - wordPressComLabelSize.width - 5,
+                                              (_siteAddressField.frame.size.height - wordPressComLabelSize.height) / 2 - 1,
+                                              wordPressComLabelSize.width,
+                                              wordPressComLabelSize.height);
+    
+    // Layout Create Blog Button
+    x = (viewWidth - CreateBlogButtonWidth)/2.0;
+    y = CGRectGetMaxY(_siteAddressField.frame) + CreateBlogStandardOffset;
+    _createBlogButton.frame = CGRectIntegral(CGRectMake(x,
+                                                           y,
+                                                           CreateBlogButtonWidth,
+                                                           CreateBlogButtonHeight));
+    
+    NSArray *controls = @[_titleLabel, _siteTitleField,
+                          _createBlogButton, _siteAddressField];
+    [WPNUXUtility centerViews:controls withStartingView:_titleLabel andEndingView:_createBlogButton forHeight:viewHeight];
+}
+
+- (IBAction)cancelButtonAction
+{
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (IBAction)viewWasTapped:(UITapGestureRecognizer *)gestureRecognizer
+{
+    [self.view endEditing:YES];
+}
+
+- (IBAction)createBlogButtonAction
+{
+    [self.view endEditing:YES];
+    
+    if (![self fieldsValid]) {
+        [self showFieldsNotFilledError];
+        return;
+    }
+    
+    [self createUserAndSite];
+}
+
+- (void)keyboardWillShow:(NSNotification *)notification
+{
+    NSDictionary *keyboardInfo = notification.userInfo;
+    CGFloat animationDuration = [[keyboardInfo objectForKey:UIKeyboardAnimationDurationUserInfoKey] floatValue];
+    CGRect keyboardFrame = [[keyboardInfo objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
+    keyboardFrame = [self.view convertRect:keyboardFrame fromView:nil];
```

Comment:
@kwonye might be best to convert the `keyboardFrame` via `[self.view.window convertRect:keyboardFrame toView:self.view]` for a more accurate conversion as to where the `keyboardFrame` is actually sourced from.